library/sha256.c

```diff
@@ -45,6 +45,14 @@
 #endif /* MBEDTLS_PLATFORM_C */
 #endif /* MBEDTLS_SELF_TEST */
 
+#if defined(MBEDTLS_PLATFORM_C)
+#include "mbedtls/platform.h"
+#else
+#include <stdlib.h>
+#define mbedtls_calloc    calloc
+#define mbedtls_free       free
+#endif
```

Comment:
This seems (almost) a duplicate of the preprocessor directives in lines 40-45. Wouldnt it be better to just add the missing lines there. Also, I am not sure you even need to remove the `#if defined(MBEDTLS_SELF_TEST)` part because the function modified in this file is a `self_test`?